### PR TITLE
Trace based sampling for logs

### PR DESCRIPTION
**Pull request by lzchen**
From [specs](https://github.com/aep-health-and-standards/Telemetry-Collection-Spec/blob/main/OpenTelemetry/logs/TraceBasedSampling.md).

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py [hunk 1]
```diff
@@ -69,6 +69,7 @@ def configure_azure_monitor(**kwargs) -> None:  # pylint: disable=C4758
      `{"azure_sdk": {"enabled": False}, "flask": {"enabled": False}, "django": {"enabled": True}}`
      will disable Azure Core Tracing and the Flask instrumentation but leave Django and the other default
      instrumentations enabled.
+    :keyword list[~opentelemetry.sdk.resources.Resource] resource: OpenTelemetry `Resource` used for the SDK.
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py:72]: Resource PR is not merged yet.

File: sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py [hunk 2]
```diff
@@ -124,7 +125,7 @@ def _setup_logging(configurations: Dict[str, ConfigurationValue]):
     logger_provider = LoggerProvider(resource=resource)
     set_logger_provider(logger_provider)
     log_exporter = AzureMonitorLogExporter(**configurations)
-    log_record_processor = BatchLogRecordProcessor(
+    log_record_processor = _AzureMonitorLogRecordProcessor(
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py:128]: Need DisableTraceBasedSamplingForLogs option.

Comment (jeremydvoss) [reply sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py:128]: Vishwesh says no env var is planned. Only a param. How about "disable_trace_based_sampling_for_logs"?

File: sdk/monitor/azure-monitor-opentelemetry/tests/processor/test_processor.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    def test_emit_disabled(self, emit_mock):
+        exporter = Mock()
+        log_data = Mock()
+        log_record = Mock()
+        log_record.span_id = 123
+        log_record.trace_flags = TraceFlags(TraceFlags.DEFAULT)
+        log_data.log_record = log_record
+        proc = _AzureMonitorLogRecordProcessor(exporter, disable_trace_based_sampling=True)
+        proc.emit(log_data)
+        emit_mock.assert_called_once_with(log_data)
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry/tests/processor/test_processor.py:50]: Nit: tests for edge cases like when trace flags don't exist

File: sdk/monitor/azure-monitor-opentelemetry/tests/configuration/test_configure.py [hunk 4]
```diff
@@ -300,11 +300,11 @@ def test_setup_logging(
...
-        blrp_mock.assert_called_once_with(
+        processor_mock.assert_called_once_with(
             log_exp_init_mock,
         )
         lp_init_mock.add_log_record_processor.assert_called_once_with(
-            blrp_init_mock
+            processor_init_mock
         )
         logging_handler_mock.assert_called_once_with(
             logger_provider=lp_init_mock
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry/tests/configuration/test_configure.py:310]: Tests for disabling trace based log sampling

**Comment by lzchen** [comment]
Waiting on this pr: https://github.com/aep-health-and-standards/Telemetry-Collection-Spec/pull/246, do not merge for now.
